### PR TITLE
several new QoL changes on Tileset editor

### PR DESCRIPTION
**Pull request by shinespeciall**
Add info textbox in Tileset Editor to show unused palettes
Now you can import Tile16 with the same way you import Tile8x8
more update on the same branch: Press BSP or DEL to delete selected Tile16 Tileset Editor
Support cleanup duplicated Tile8x8 from the Tile8x8 set and Tile16 set in Tileset Editor

### CODE REVIEW

File: Dialog/TilesetEditDialog.cpp [hunk 1]
```diff
@@ -581,6 +591,28 @@ void TilesetEditDialog::OverwriteATile8x8InTile8x8MapAndUpdateTile16Map(int posI
...
+/// Update Info Textbox
+/// </summary>
+void TilesetEditDialog::UpdateInfoTextBox()
+{
+    // clean up textbox
+    ui->textEdit_Infos->setText("");
+
+    // Generate new Infos
+    QString output;
+    output += tr("Unused Palette(s): (don't forget to clean up those trash Tile16 manually)\n");
```
Comment (Goldensunboy) [inline Dialog/TilesetEditDialog.cpp:604]: trash -> unused

File: Dialog/TilesetEditDialog.cpp [hunk 2]
```diff
@@ -993,3 +932,272 @@ void TilesetEditDialog::on_pushButton_ImportPalette_clicked()
...
+     *  don't overwrite the first Tile16
+     *  don't let imported Tile16 id goes out of bound (should be smaller than 0x2FF)
+     *  TODO: support flexible Tile16 set length in the future, and we can read each of its length from the RATS struct
+    **/
+    LevelComponents::Tileset *tmp_newTilesetPtr = tilesetEditParams->newTileset;
+    int selTile16 = SelectedTile16;
+    int selPalId = SelectedPaletteId;
+    FileIOUtils::ImportTile8x8GfxData(this,
+        tmp_newTilesetPtr->GetPalettes()[SelectedPaletteId],
+        [selTile16, tmp_newTilesetPtr, selPalId] (QByteArray finaldata, QWidget *parentPtr)
```
Comment (Goldensunboy) [inline Dialog/TilesetEditDialog.cpp:951]: Try to consolidate this code in FileIOUtils from the multiple callbacks that use it.

File: ROMUtils.cpp [hunk 3]
```diff
@@ -146,6 +146,45 @@ namespace ROMUtils
...
+    /// <summary>
+    /// Get an 32 bytes uchar array of Tile8x8 graphic data, then X flip the data of its graphic.
+    /// </summary>
+    /// <param name="source">
+    /// The pointer to read uchar array Tile8x8 data.
+    /// </param>
+    /// <param name="destination">
+    /// The pointer to save changed Tile8x8 data to.
+    /// </param>
+    void Tile8x8DataXFlip(unsigned char *source, unsigned char *destination)
```
Comment (Goldensunboy) [inline ROMUtils.cpp:158]: Is there perhaps a scenario where we want to read the tile with no flip? Or perhaps X and Y flip? This seems like something that should be one function with bool xFlip, bool yFlip. What do you think about this?

File: Dialog/TilesetEditDialog.cpp (hunk 1 above)
Comment (shinespeciall) [reply Dialog/TilesetEditDialog.cpp:604]: “Unused Palette(s): (don't forget to clean up those unused Tile16 to free up the room for new palettes.”

File: Dialog/TilesetEditDialog.cpp (hunk 2 above)
Comment (shinespeciall) [reply Dialog/TilesetEditDialog.cpp:951]: so the first one find if we can import Tile16 into the Tileset by comparing each existing Tile8x8, and see if the elements in Tile16 match one of them
the second one eliminate replicated Tile8x8 and managing the Tile16 all using the Tile8x8 id in the new arranged Tile8x8 set (edited)
they just looks similar, but not function similarly
(record from private talk)

**Review by Goldensunboy — APPROVED** [review]

**Pull request merged**